langchain4j-vertex-ai/src/main/java/dev/langchain4j/store/embedding/vertexai/internal/IndexEndpointService.java

```diff
@@ -0,0 +1,153 @@
+package dev.langchain4j.store.embedding.vertexai.internal;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.cloud.aiplatform.v1.*;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.experimental.SuperBuilder;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static dev.langchain4j.internal.Utils.isNullOrBlank;
+
+@Slf4j
+@SuperBuilder
+public class IndexEndpointService {
+
+    @NonNull
+    private final String endpoint;
+    @NonNull
+    private final String location;
+    @NonNull
+    private final String project;
+    private final String indexEndpointId;
+    private final String indexId;
+    @Getter(lazy = true)
+    private final IndexServiceSettings indexServiceSettings = initIndexServiceClientSettings();
+    private final CredentialsProvider credentialsProvider;
+    @Getter(lazy = true)
+    private final String publicEndpoint = initPublicEndpoint();
+
+    /**
+     * Deletes the embedding index.
+     *
+     * @param embeddingIndices the embedding indices
+     */
+    public void deleteIndices(List<String> embeddingIndices) {
+        if (isNullOrBlank(indexId)) {
+            log.warn("Index is not specified, skipping delete.");
+            return;
+        }
+
+        IndexName name = IndexName.of(project, location, indexId);
+
+        final RemoveDatapointsRequest request = RemoveDatapointsRequest.newBuilder()
+                .setIndex(name.toString())
+                .addAllDatapointIds(embeddingIndices)
+                .build();
+
+        try (IndexServiceClient indexServiceClient = IndexServiceClient.create(getIndexServiceSettings())) {
+            indexServiceClient.removeDatapoints(request);
+        } catch (IOException exception) {
+            log.error("Failed to create IndexServiceClient", exception);
+            throw new RuntimeException("Failed to create IndexServiceClient", exception);
+        }
+    }
+
+    /**
+     * Upsert the embedding index.
+     *
+     * @param embeddingIndex the embedding index
+     */
+    public void upsertEmbedding(VertexAiEmbeddingIndex embeddingIndex) {
+        if (isNullOrBlank(indexId)) {
```

Comment:
Why is this check required? Creating `IndexEndpointService` without `indexId` should fail

Reply:
Yes, just did that for convenience but happy to make it failed.